src/App.test.tsx

```diff
@@ -49,14 +53,47 @@ describe('loads election', () => {
     const { getByText } = render(<App />)
     getByText('Configure Ballot Marking Device')
   })
+  it(`App uses data from localStorage when set`, () => {
+    const ballotStyleId = electionSample.ballotStyles[0].id
+    const precinctId = electionSample.precincts[0].id
+    const presidentFirstCandidate = electionSample.contests.find(
+      c => c.id === 'president'
+    )!.candidates![0]
+
+    window.localStorage.setItem(electionKey, electionSampleAsString)
+    window.localStorage.setItem(
+      activationStorageKey,
+      JSON.stringify({
+        ballotStyleId,
+        precinctId,
+      })
+    )
+    window.localStorage.setItem(
```

Comment:
It's very risky to store the voter's choice in `localStorage`. Too many ways that gets written to disk, not cleaned up. Same goes for the activation code, which might eventually include a signature. I feel pretty strongly that these should be only in memory, and should go away fairly aggressively if there's any reload / error / etc.